server/gfmd/user.c

```diff
@@ -183,13 +160,35 @@ gfarm_error_t
 user_auth_id_modify(struct user *u, const char *auth_id_type,
 	const char *auth_user_id)
 {
-	return (GFARM_ERR_NO_ERROR);
+	enum auth_user_id_type auth_user_id_type = 0;
+	struct db_user_auth_arg arg = {
+		u->ui.username,
+		(char *)auth_id_type,
+		(char *)auth_user_id
+	};
+	gfarm_error_t e;
+
+	if ((e = gfarm_auth_user_id_type_from_name(
+			(char *)auth_id_type,
+			&auth_user_id_type)) == GFARM_ERR_NO_ERROR) {
+		if (u->auth_user_id[auth_user_id_type] == NULL) {
+			return (db_user_auth_add(&arg));
+		} else {
+			return (db_user_auth_modify(&arg));
+		}
+	}
+
+	return (e);
 }
 
 gfarm_error_t
 user_auth_id_remove(struct user *user, const char *auth_id_type)
 {
-	return (GFARM_ERR_NO_ERROR);
```

Comment:
The user_auth_id_remove() function should update u->auth_user_id[] and user_id_map too.